plugins/camspipeline.py

```
@@ -0,0 +1,200 @@
+        existing = pd.read_sql(
+            f"SELECT time, latitude, longitude FROM {table_name}",
+            engine
+        )
+        if not existing.empty:
+            df['key'] = df[['time', 'latitude', 'longitude']].apply(tuple, axis=1)
+            existing['key'] = existing[['time', 'latitude', 'longitude']].apply(tuple, axis=1)
+            new_records = df[~df['key'].isin(existing['key'])].drop(columns='key')
+            return new_records
+        return df
+    except SQLAlchemyError as e:
+        raise RuntimeError(f"Failed to check duplicates for {table_name}: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)